explorer/templates_test.go

```diff
@@ -0,0 +1,59 @@
+package explorer
+
+import (
+	"testing"
+)
+
+func TestamountAsDecimalPartsTrimmed(t *testing.T) {
+	test1 := amountAsDecimalPartsTrimmed(314159000, 2, false)
+	test2 := amountAsDecimalPartsTrimmed(76543210000, 2, false)
+	test3 := amountAsDecimalPartsTrimmed(76643210000, 2, false)
+	test4 := amountAsDecimalPartsTrimmed(654321000, 1, false)
+	test5 := amountAsDecimalPartsTrimmed(987654321, 8, false)
+	test6 := amountAsDecimalPartsTrimmed(987654321, 2, false)
+	test7 := amountAsDecimalPartsTrimmed(90765432100, 2, false)
+	test8 := amountAsDecimalPartsTrimmed(9076543200000, 2, false)
+	test9 := amountAsDecimalPartsTrimmed(907654320, 7, false)
+	test10 := amountAsDecimalPartsTrimmed(1234590700, 2, false)
+
+	if test1[0]+"."+test1[1] != "3.14" {
+		t.Error("Wrong output, got" + test1[0] + "." + test1[1])
+	}
+
+	if test2[0]+"."+test2[1] != "765.43" {
```

Comment:
Please rewrite this test with an array of input data and expected outputs.  Along these lines:

```go
	in := []struct {
		amt    int64
		n      int64
		commas bool
	}{
		{314159000, 2, false},
		{76543210000, 2, false},
	}

	expected := []struct {
		whole, frac, tail string
	}{
		{"3", "14", ""},
		{"765", "43", ""},
	}

	for i := range in {
		out := amountAsDecimalPartsTrimmed(in[i].amt, in[i].n, in[i].commas)
		if out[0] != expected[i].whole || out[1] != expected[i].frac ||
			out[2] != expected[i].tail {
			t.Errorf("amountAsDecimalPartsTrimmed failed for "+
				"%d (%d decimals, commas=%v). Got %s.%s%s, expected %s.%s%s.",
				in[i].amt, in[i].n, in[i].commas,
				out[0], out[1], out[2],
				expected[i].whole, expected[i].frac, expected[i].tail)
		}
	}
```